Tupelo naming service

cap10morgan opened the pull request:
This seems like a useful primitive to build on top of some more. But if others disagree I won't be offended. :)

Having a fast, reliable, programmable, decentralized(-ish and less -ish in the future) mutable pointer system seems pretty cool.

What else needs to happen with this now that the demo for which it was created is over but before it's ready to merge to master?

cap10morgan committed:
Random tiny formatting cleanups

cap10morgan committed:
Move Objects tab next to Creator

...to create a better visual association between them

cap10morgan committed:
Add initial TNS tab, form, and entries table

cap10morgan committed:
Add basic TNS router component

cap10morgan committed:
Use @ name for root like DNS

cap10morgan committed:
Get TNS redirects working & add loader for them

cap10morgan committed:
Clarify that TNS entry form can also modify existing

cap10morgan committed:
Start on Tokens tab

brandonwestcott reviewed (approved):
I'm 👍 👍 for merging in the functionality, I think its a great example of a completely comprehendible, but unique use of NFTs

The only modification I would suggest is removing dependence on `clownfahrt.com` and instead using either subdomains or paths on `tupelo.me`

cap10morgan committed:
Allow per-environment config of wallet domain

This is the domain that will show the wallet UI rather than looking up a
wallet for TNS-based redirects.

cap10morgan committed:
Support configured WALLET_DOMAIN and subdomains

cap10morgan committed:
Make IPFS gateway configurable

cap10morgan commented:
@brandonwestcott Changes since your initial review look OK? Good suggestion. I had intended to do that but totally forgot. Now it supports subdomains too (i.e. `wesmorgan.tupelo.me` will look up the `@` TNS record in the `wesmorgan` wallet while a CNAME'd domain will still look up that domain as the wallet name and load that record.

cap10morgan committed:
Fix off-by-one error in CNAME subdomain TNS lookup
